evalml/tests/automl_tests/test_automl.py

```diff
@@ -1915,13 +1973,20 @@ class DummyPipeline(BinaryClassificationPipeline):
         num_pipelines_different_seed = 0
         num_pipelines_init = 0
 
-        def __init__(self, parameters, random_seed):
+        def __init__(self, parameters, random_seed=0):
             is_diff_random_seed = not (random_seed == 42)
             self.__class__.num_pipelines_init += 1
             self.__class__.num_pipelines_different_seed += is_diff_random_seed
-            super().__init__(parameters, random_seed=random_seed)
+            super().__init__(self.component_graph, parameters=parameters, random_seed=random_seed)
 
-    automl = AutoMLSearch(X_train=X, y_train=y, problem_type="binary", allowed_pipelines=[DummyPipeline],
+        def new(self, parameters, random_seed):
+            return self.__class__(parameters, random_seed=random_seed)
+
+        def clone(self):
+            return self.__class__(self.parameters, random_seed=self.random_seed)
+    pipelines = [DummyPipeline({})]
+    DummyPipeline.num_pipelines_different_seed = 0
```

Comment:
This test is interesting, lol. It tries to check whether or not pipelines instantiated in AutoMLSearch use automl's random seed or not. We need to instantiate a pipeline instance to pass to AutoMLSearch, causing `num_pipelines_different_seed` to increase to 1. I'm setting here back to 0, so that we know the calls are strictly from automl. Alternatively, could have checked for 1 non-42-pipeline but I think this is slightly clearer.